operators/src/mock/mock_raster_point_join.rs

```diff
@@ -62,91 +63,101 @@ where
-        params: MockRasterPointJoinParams,
-    ) -> MockRasterPointJoinProcessor<
-        Box<dyn RasterQueryProcessor<RasterType = T1>>,
-        Box<dyn VectorQueryProcessor<VectorType = MultiPointCollection>>,
-    >
-    where
-        T1: Copy + Sync + 'static,
-    {
-        MockRasterPointJoinProcessor::new(source_a, source_b, params)
-    }
+fn create_binary_raster_vector<TR, TV>(
+    source_a: Box<dyn RasterQueryProcessor<RasterType = TR>>,
+    source_b: Box<dyn VectorQueryProcessor<VectorType = TV>>,
+    params: MockRasterPointJoinParams,
+) -> MockRasterPointJoinProcessor<
+    Box<dyn RasterQueryProcessor<RasterType = TR>>,
+    Box<dyn VectorQueryProcessor<VectorType = TV>>,
+>
+where
+    TR: Copy + Sync + 'static,
+{
+    MockRasterPointJoinProcessor::new(source_a, source_b, params)
 }
 
 #[derive(Debug, Serialize, Deserialize, Clone)]
 pub struct MockRasterPointJoinParams {
     pub feature_name: String,
 }
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct MockRasterPointJoinOperator {
-    raster_sources: Vec<Box<dyn RasterOperator>>,
-    point_sources: Vec<Box<dyn VectorOperator>>,
-    params: MockRasterPointJoinParams,
-}
-
-impl Operator for MockRasterPointJoinOperator {
-    fn raster_sources(&self) -> &[Box<dyn RasterOperator>] {
-        &self.raster_sources
-    }
-    fn vector_sources(&self) -> &[Box<dyn VectorOperator>] {
-        &self.point_sources
-    }
-    fn projection(&self) -> ProjectionOption {
-        self.point_sources
-            .get(0)
-            .map_or_else(|| ProjectionOption::None, |o| o.projection())
-    }
-}
+pub type MockRasterPointJoinOperator = OperatorImpl<MockRasterPointJoinParams>;
 
 #[typetag::serde]
 impl VectorOperator for MockRasterPointJoinOperator {
-    fn result_type(&self) -> VectorDataType {
-        VectorDataType::MultiPoint
+    fn into_initialized_operator(
+        self: Box<Self>,
+        context: crate::engine::ExecutionContext,
+    ) -> Result<Box<dyn crate::engine::InitializedVectorOperator>> {
+        InitilaizedOperatorImpl::create(
+            self.params,
+            context,
+            |_, _, _, vs| {
+                Ok(VectorResultDescriptor {
+                    projection: vs.get(0).map_or_else(
+                        || ProjectionOption::None,
+                        |o| o.result_descriptor().projection,
+                    ),
+                    data_type: VectorDataType::MultiPoint,
+                })
+            },
+            self.raster_sources,
+            self.vector_sources,
+        )
+        .map(InitilaizedOperatorImpl::boxed)
     }
+}
 
+impl InitializedVectorOperator
+    for InitilaizedOperatorImpl<MockRasterPointJoinParams, VectorResultDescriptor>
+{
     fn vector_processor(&self) -> Result<crate::engine::TypedVectorQueryProcessor> {
-        self.validate_children(1..2, 1..2)?;
+        // self.validate_children(self.result_descriptor().projection, 1..2, 1..2)?;
```

Comment:
Why is this commented out?

Reply:
i guess validation should be done in the initialized_operator construction step. This should be deleted.